data/blog/2022-11-07.retreat.md

```diff
@@ -0,0 +1,95 @@
+
+### Day to day activities
+
+Apart from talking to old and new friends face-to-face, which we haven't done for some time, we also worked on various projects to improve the MirageOS ecosystem. Below is a collection of projects that wouldn't come to existance without the retreat. Please excuse if there's some project missing.
+
+In the day time we discussed various topics, including performance considerations and how to write protocol implementations with GADTs. Next to that people were going to the beach, while others were pairing up for some projects or reading code together. Others worked alone on finishing some code, but whenever a question of another library appeared, it was easy to approach the original author or someone who has used that library extensively and discuss some questions. In the evenings we had some impromptu talks about various topics, such as the opam repository CI, Lwt do's and dont's, BGP, monitoring, PWM.
+
+One afternoon we did a long [dragon dreaming](https://dragondreaming.org/) session to figure out the ideas and dreams for the future of MirageOS. Apart from bare-metal and MirageOS as hypervisor, we also dreamed about running your own digital infrastructure with MirageOS (with increased decentralization and security). Another afternoon we had a cocktail party at the pool.
+
+Some participants already wrote articles about their journey and activities, take a look at their writings: [Raphaël Proust](https://raphael-proust.gitlab.io/code/mirage-retreat-2022-10.html), [Jules, Sayo, Enguerrand, Sonja, Jan, Lucas](https://tarides.com/blog/2022-10-28-the-mirageos-retreat-a-journey-of-food-cats-and-unikernels), [Pierre](http://blog.enssat.fr/2022/10/pierre-alain-enssat-teacher-at-11th.html).
+
+#### OCaml 5 support
+
+OCaml 5 will soon be published, which has most of the runtime code being rewritten. The runtime also has many more dependencies. We worked on getting [ocaml-solo5](https://github.com/mirage/ocaml-solo5/) (the piece consisting of a math library, a C library, and the OCaml compiler for MirageOS unikernels) to use [OCaml 5](https://github.com/mirage/ocaml-solo5/pull/122). This is still work in progress, but already lead to [some](https://github.com/ocaml/ocaml/pull/11605) [upstream](https://github.com/ocaml/ocaml/pull/11606) [PRs](https://github.com/ocaml/ocaml/pull/11611) that are merged into the 5 branch as well.
+
+#### Integrating a Verified File System
+
+[FSCQ](https://github.com/mit-pdos/fscq) is a formally verified file system developed in the Coq proof assistant. So far MirageOS has a variety of [chamelon](https://github.com/yomimono/chamelon/) (LittleFS) [tar](https://github.com/mirage/ocaml-tar/) (append-only), and [fat](https://github.com/mirage/ocaml-fat/) (FAT16, barely used), [one file FS](https://github.com/reynir/oneffs) (supporting only one file).
```

Comment:
This second sentence was a bit hard for me to read. Perhaps rephrase it to
`So far MirageOS has a variety: [...]` or
`So far MirageOS has a variety of file system options: [...]`

Reply:
better now (eecf326)?

Reply:
Yep, thanks!